### PR TITLE
add delete nostr profile ability

### PR DESCRIPTION
**Pull request by futurepaul**
needs https://github.com/MutinyWallet/mutiny-node/pull/1096

### CODE REVIEW

File: src/routes/settings/NostrKeys.tsx [hunk 1]
```diff
@@ -18,12 +20,57 @@ import {
...
+    }
+
+    const [confirmOpen, setConfirmOpen] = createSignal(false);
+    const [confirmLoading, setConfirmLoading] = createSignal(false);
+
+    async function deleteNostrAccount() {
+        setConfirmLoading(true);
+        try {
+            await state.mutiny_wallet?.delete_profile();
+            console.log("yooo");
```
Comment (benthecarman) [inline src/routes/settings/NostrKeys.tsx:38]: don't need tihs console log

File: src/routes/settings/NostrKeys.tsx [hunk 2]
```diff
@@ -18,12 +20,57 @@ import {
...
+        setConfirmOpen(true);
+    }
+
+    const [confirmOpen, setConfirmOpen] = createSignal(false);
+    const [confirmLoading, setConfirmLoading] = createSignal(false);
+
+    async function deleteNostrAccount() {
+        setConfirmLoading(true);
+        try {
+            await state.mutiny_wallet?.delete_profile();
```
Comment (benthecarman) [inline src/routes/settings/NostrKeys.tsx:37]: should delete the local nsec if one is imported too

**Review by benthecarman — APPROVED** [review]
works for me

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
Importing an nsec after deleting the main wallet nsec doesn't work.

**Comment by futurepaul** [comment]
> Importing an nsec after deleting the main wallet nsec doesn't work.

It works for me. I made a new profile with a new wallet, deleted the profile, and then imported my normal nsec.

File: src/routes/Profile.tsx [hunk 3]
```diff
@@ -42,11 +42,21 @@ export function Profile() {
...
     const [copy, copied] = useCopy({ copiedTimeout: 1000 });
 
+    const profileDeleted = createMemo(() => {
+        // return true if profile.deleted is true or "true"
+        // return false if profile.deleted is false or "false"
+        if (profile().deleted === true || profile().deleted === "true") {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (benthecarman) [inline src/routes/Profile.tsx:52]: this can just be `(profile().deleted === true || profile().deleted === "true")` no need for the if statement

**Review by benthecarman — APPROVED** [review]
Paul and I tested every case we could think of on a call, this is good to go

**Review by TonyGiorgio — APPROVED** [review]

**Review by TonyGiorgio — APPROVED** [review]

**Pull request merged**